src/main.c

```diff
@@ -1706,6 +1706,14 @@ static int open_inet_listener(h2o_configurator_command_t *cmd, yoml_t *node, con
     return fd;
 }
 
+static void set_socket_buffers(int fd, unsigned sndbuf, unsigned rcvbuf)
+{
+    if (sndbuf != 0 && setsockopt(fd, SOL_SOCKET, SO_SNDBUF, &sndbuf, sizeof(sndbuf)) != 0)
+        h2o_fatal("failed to set SO_SNDBUF:%s", strerror(errno));
```

Comment:
`set_socket_buffers` has to be MT-safe when called from `h2o_accept`. Therefore, we cannot use `strerror`. `h2o_strerror` should be used instead.